ic-asset/src/sync.rs

```diff
@@ -56,32 +59,44 @@ fn filename_starts_with_dot(entry: &walkdir::DirEntry) -> bool {
         .unwrap_or(false)
 }
 
-fn gather_asset_locations(dirs: &[&Path]) -> anyhow::Result<Vec<AssetLocation>> {
-    let mut asset_descriptors: HashMap<String, AssetLocation> = HashMap::new();
+fn gather_asset_descriptors(dirs: &[&Path]) -> anyhow::Result<Vec<AssetDescriptor>> {
+    let mut asset_descriptors: HashMap<String, AssetDescriptor> = HashMap::new();
     for dir in dirs {
-        let asset_locations = WalkDir::new(dir)
+        let configuration = AssetSourceDirectoryConfiguration::load(dir)?;
+        let asset_descriptors_interim = WalkDir::new(dir)
             .into_iter()
             .filter_entry(|entry| !filename_starts_with_dot(entry))
             .filter_map(|r| {
                 r.ok().filter(|entry| entry.file_type().is_file()).map(|e| {
                     let source = e.path().to_path_buf();
                     let relative = source.strip_prefix(dir).expect("cannot strip prefix");
                     let key = String::from("/") + relative.to_string_lossy().as_ref();
-
-                    AssetLocation { source, key }
+                    let config = configuration
+                        .get_asset_config(&source)
+                        .context(format!(
+                            "failed to get config for asset: {}",
+                            source.to_str().unwrap()
+                        ))
+                        .unwrap(); // TODO
```

Comment:
I'm struggling to find a good way to bail out of this closure when we can't find the config, do you have an idea that could help here? Perhaps we should not use the iterator and rely on for-loop instead?